mimicpy/core/prepare.py

```diff
@@ -55,43 +55,52 @@ def clear(self):
         """Empty the QM region to start over"""
         self.qmatoms = pd.DataFrame()
     
-    def getInp(self, ndx=None, inp_file=None):
+    def getInp(self, inp=None, mdp=None, ndx_file=None, cpmd_file=None):
         """
         Create the QM region from the atoms added
         Steps done:
-            Assign QM atoms as list to inp._ndx
-            If named fiven, writes index files of atoms in QM region with name QMatoms
+            If named given, writes index files of atoms in QM region with name QMatoms
             Sort self.qmatoms by link and elements
             Read element symbols and all charge info from self.qmatoms
             Add MIMIC, SYSTEM, DFT sections of CPMD script
             Add all atoms to CPMD script
         """
+        
+        if isinstance(mdp, str):
+            mdp = MDP.fromFile(mdp)
+        
+        maxsteps, timestep, mdp_errors = _qmhelper.check_mdp(mdp)
```

Comment:
Catch error if something is wrong with mdp file

Reply:
Are you referring to the mdp_errors variable? If so, that will printed out and for the user to go through and to edit his MDP file accordingly before actually running mimic.

On that note, the mdp error checking is done in check_mdp(). We will have to add more error checking, in consultation with Slava.

Reply:
More like a FileNotFound situation

Reply:
Not exactly, it's error messages that we should display to the user. Like if the mdp file didn't have an integrator = mimic, we will display the message please add intergator=mimic in your input file.

So, it's not really an error with executing of the python script, but errors that could come when running the actual mimic thing. MiMiCPy will try and anticipate those errors.